### ISSUE TITLE
Require mentor's approval for task completion

### ISSUE THREAD

**Issue by bartekpacia** [anitab-org/mentorship-backend#278]
**Is your feature request related to a problem? Please describe.**
This was first raised by [asawesome07](https://github.com/asawesome07) in [this PR](https://github.com/systers/mentorship-android/issues/335).
What he suggests is adding a new feature: in a specific mentorship relatio, the user can't mark the task as completed himself - he needs the mettor's approval.

**Describe the solution you'd like**
I think the best solution is to add a new property to relation data model. This property would indentify whether mentee can mark tasks as completed by himself or needs a mentor to approve task completion.

This property could be named [for example] `requires_approval: [boolean]`

It would default to `false` when the relation is created. After that, only mentor would be able to change it.

**Describe alternatives you've considered**
I think the above is the simplest solution, but there might be a better one I'm not aware of.

**Comment by isabelcosta** [comment]
Good idea @bartekpacia  🎉 I think that has crossed my mind during the GSoC program, but for the sake of having a minimum functional product, I think we decided to simplify there. This is a great idea to think now :)

There could be something added to the task model. Here's what is exposed by the API:
```
{
  "id": 0,
  "description": "string",
  "is_done": true,
  "created_at": 0,
  "completed_at": 0
}
```
To facilitate brainstorming this issue

**Comment by bartekpacia** [comment]
Great, I'll try to look into in when I finish implementing CLI :)

**Comment by bartekpacia** [comment]
Aaaand...done:D

**Comment by vermastuti** [comment]
Is this issue resolved?

**Comment by bartekpacia** [comment]
I did it long time ago in #292, but can't bring that branch up to date with master right now

**Comment by RiddhiAthreya** [comment]
@vj-codes can I work on this?

**Comment by vj-codes** [comment]
Assigning you @RiddhiAthreya 
Happy coding!
